graphql-codegen/codegen.ts

```diff
@@ -0,0 +1,52 @@
+import { addTypenameSelectionDocumentTransform } from '@graphql-codegen/client-preset'
+import type { CodegenConfig } from '@graphql-codegen/cli'
+
+const config: CodegenConfig = {
+  schema: 'http://127.0.0.1:54321/graphql/v1',
+  overwrite: true,
+  watch: true,
+  documents: ['app/**/*.{ts,tsx,graphql}'],
+  generates: {
+    'graphql-codegen/generated/': {
+      preset: 'client',
+      presetConfig: {
+        persistedDocuments: true,
+        apolloClientVersion: 3,
+      },
+      documentTransforms: [addTypenameSelectionDocumentTransform],
+    },
+    // NOTE: api.ts generates automatically custom hooks for each query/mutation
+    'graphql-codegen/generated/api.ts': {
+      overwrite: true,
+      plugins: ['typescript', 'typescript-operations', 'typescript-react-apollo'],
+      config: {
+        // NOTE: https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-react-apollo
+        useTypeImports: true,
+        withComponent: false,
+        withHOC: false,
+        withHooks: true,
+        withRefetchFn: true,
+      },
+      documentTransforms: [addTypenameSelectionDocumentTransform],
+    },
+  },
+  ignoreNoDocuments: true,
+  config: {
+    useTypeImports: true,
+    scalars: {
+      UUID: 'string',
+      Date: 'string',
+      Time: 'string',
+      Datetime: 'string',
+      JSON: 'string',
+      BigInt: 'number',
```

Comment:
Probably this has to do with the BIGSERIAL which is PostgreSQL data type though, it wouldn't safely store data, we might wanna use `bigint` of JS type instead of `number`.

But what's more, we might not wanna use BIGSERIAL in frontend, instead we might wanna use UUID. Detail: above comment.

Reply:
I agree with your opinion.
Handling Bigint on the frontend can be a bit cumbersome, so it might be better to use UUIDs instead.

The problem is also range of `pg_graphql`, I think. 🤔